### PR TITLE
166214110 Super-admin creates, activate or deactivate a user

### PR DESCRIPTION
**Pull request by Lundii**
#### What does this PR do?
It allows a super admin to create or delete a new user
#### Description of Task to be completed?
* created a route to create a user
* created a route to deactivate or activate a user
#### How should this be manually tested?
To create a user
* pull the branch
* run npm install to download all packages
* run `npm run db:roll` to create the tables
* run `npm run db:seed` to load the seeder files
* login as a superadmin using the login route `POST /api/v1/auth/login` using the details
  ``` 
    email: hellobooks@email.com
    password: password
  ```
* create a new user using the route `POST /api/v1/admin/user` with the following details
 ```
   firstName: .......
   lastName: .........
   email: ..........
   role: ........
 ```
   The available roles on the seeder file include
        1.  user
        2.  admin
        3.  patron
   You can always add yours
* login with the newly created user using the email and the password in your .env file.
* a mail is sent to the newly created user with the password

To deactivate or activate a user, visit
 * `PATCH/api/v1/admin/user/userId` as a superadmin to deactivate or a activate a registered user
#### Any background context you want to provide?
No
#### What are the relevant pivotal tracker stories?
[#166214110](https://www.pivotaltracker.com/story/show/166214110)
#### Screenshots (if appropriate)
No
#### Questions:
No

### CODE REVIEW

File: server/controllers/AdminController.js [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+   * @param {object} response
+   * @returns {json} json
+   * @memberof AdminController
+   */
+  static async changeUserStatus(request, response) {
+    const { id } = request.params;
+    const { status } = request.body;
+    try {
+      const user = await User.findOne({ where: { id } });
+      if (!user) {
```
Comment (KvNGCzA) [inline server/controllers/AdminController.js]: You can call the `findUser` helper function here, then you can save space by getting rid of `line 29 and 30`

File: server/controllers/AuthController.js [hunk 2]
```diff
@@ -22,25 +23,35 @@ export default class AuthController {
...
+    let newUserFromAdmin;
     try {
+      if (!password) {
+        newUserFromAdmin = true;
+      }
       const existingUser = await User.findOne({ where: { email } });
       if (existingUser) return responseMessage(response, 409, { message: 'user already exist' });
       const newUser = await User.create({
-        firstName, lastName, email, password: bcrypt.hashSync(password, 10), avatarUrl
+        firstName, lastName, email, password: bcrypt.hashSync(password || 'password@1', 10), avatarUrl
```
Comment (KvNGCzA) [inline server/controllers/AuthController.js]: Lets make this default password a `process.env` key

**Review by KvNGCzA — CHANGES_REQUESTED** [review]

File: server/controllers/AuthController.js (hunk 2 above)
Comment (Lundii) [reply server/controllers/AuthController.js]: > Lets make this default password a `process.env` key

Done

File: server/controllers/AdminController.js (hunk 1 above)
Comment (Lundii) [reply server/controllers/AdminController.js]: > You can call the `findUser` helper function here, then you can save space by getting rid of `line 29 and 30`

Done

**Review by KvNGCzA — APPROVED** [review]

**Pull request merged**